stdlib/cmake/modules/AddSwiftStdlib.cmake

```diff
@@ -895,11 +895,48 @@ endfunction()
 
 # Add a new Swift target library.
 #
+# NOTE: This function inherits all options from add_swift_target_library_single_sdk
+# except for SDK.
+# Usage:
+#   add_swift_target_library(name
+#     [TARGET_SDKS sdk1...]
+#     [add_swift_target_library_single_sdk's options ...])
+#
+# TARGET_SDKS
+#   The set of SDKs in which this library is included. If empty, the library
+#   is included in all SDKs.
+function(add_swift_target_library name)
+  set(SWIFTLIB_multiple_parameter_options TARGET_SDKS)
+  cmake_parse_arguments(SWIFTLIB
+                        ""
+                        ""
+                        "${SWIFTLIB_multiple_parameter_options}"
+                        ${ARGN})
+
+  # If target SDKs are not specified, build for all known SDKs.
+  if("${SWIFTLIB_TARGET_SDKS}" STREQUAL "")
+    set(SWIFTLIB_TARGET_SDKS ${SWIFT_SDKS})
+  endif()
+  list_replace(SWIFTLIB_TARGET_SDKS ALL_APPLE_PLATFORMS "${SWIFT_APPLE_PLATFORMS}")
+
+  # If we are building this library for targets, loop through the various
+  # SDKs building the variants of this library.
+  list_intersect(
+      "${SWIFTLIB_TARGET_SDKS}" "${SWIFT_SDKS}" SWIFTLIB_TARGET_SDKS)
```

Comment:
Why is this needed?  Can you not just iterate over `ASTL_TARGET_SDKS`?

Reply:
This line filters out unsupported SDK variants from `ASTL_TARGET_SDKS`. This is moved from original `add_swift_target_library`.

Reply:
But `add_swift_target_library_single_sdk` also checks that given `SDK` is valid, so this is not needed 👍